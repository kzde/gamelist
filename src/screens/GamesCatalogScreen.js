import React, { Component, Fragment } from 'react';
import Header from '../components/Header';
import PlatformChoices from '../components/PlatformChoices';
import choices from '../datas/platforms.json'
import CatalogContainer from '../components/CatalogContainer';
import { getGamesByPlatform } from '../utils/selectors';
import gamesObj from '../datas/games.json';
import withI18n from '../contexts/withI18n';
import PropTypes from 'prop-types';


const { games } = gamesObj

class GamesCatalogScreen extends Component {
  constructor(props) {
    super(props);
    const defaultPlatform = choices.platforms[0];
    this.state = {
      selectedPlatform: defaultPlatform,
      games: getGamesByPlatform(games, defaultPlatform)
    }
    this.platformSelected = this.platformSelected.bind(this);
  }

  platformSelected(selectedPlatform) {
    this.setState({ 
      selectedPlatform,
      games: getGamesByPlatform(games, selectedPlatform)
    })
  }

  render() {
    return (
      <Fragment>
        <Header text={`${this.props.i18n._('catalog.header.title')} ${this.state.selectedPlatform}`} />
        <PlatformChoices
          choices={choices.platforms}
          handleClick={this.platformSelected}
          selectedPlatform={this.state.selectedPlatform}
        />
        <CatalogContainer games={this.state.games} />
      </Fragment>
    )
  }
}

GamesCatalogScreen.propsType = {
  i18n: PropTypes.object.isRequired
}
export const gamesCatalogScreen = GamesCatalogScreen;
export default withI18n(GamesCatalogScreen);