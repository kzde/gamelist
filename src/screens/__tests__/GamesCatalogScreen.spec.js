import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import GamesCatalogScreen from '../GamesCatalogScreen';

jest.mock('../../components/CatalogContainer', () => 'CatalogContainer');
jest.mock('../../components/Header', () => 'Header');
jest.mock('../../components/PlatformChoices', () => 'PlatformChoices');

describe('GamesCatalogScreen', () => {
  it('should renders correctly', () => {
    const tree = renderer
      .create(<GamesCatalogScreen />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })

  describe('platformSelected', () => {
    it('should set the selectedPlatform state when platformSelected function was called', () => {
      const wrapper = shallow(<GamesCatalogScreen />);
      wrapper.setState({selectedPlatform: 'PC'})
      wrapper.instance().platformSelected('X1');
      expect(wrapper.state('selectedPlatform')).toEqual('X1');
    })
  })
})