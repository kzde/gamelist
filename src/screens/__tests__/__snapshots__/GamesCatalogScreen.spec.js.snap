// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GamesCatalogScreen should renders correctly 1`] = `
Array [
  <Header
    text="Games catalog PS4"
  />,
  <PlatformChoices
    choices={
      Array [
        "PS4",
        "X1",
        "PC",
        "Switch",
      ]
    }
    handleClick={[Function]}
    selectedPlatform="PS4"
  />,
  <CatalogContainer
    games={
      Array [
        Object {
          "cover": "https://via.placeholder.com/359x411/003791/ffffff",
          "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi in vehicula nulla. Nullam id leo non mi gravida iaculis. Integer aliquam ex in ipsum cursus, ac accumsan nisl cursus. Curabitur quis lorem sed enim ultrices scelerisque sit amet et sapien. Integer elementum, eros vel dignissim consectetur, nibh tortor iaculis leo, nec varius dolor velit in nisl. Quisque ac nibh quis risus venenatis mollis. Aliquam dapibus, mauris nec iaculis ornare, neque nulla lacinia turpis, ac luctus nunc tortor et est. Donec tempor lorem ac eros semper, eget ultricies urna ultrices. Nunc feugiat eros dui, id pharetra risus hendrerit sit amet.",
          "hero": "https://via.placeholder.com/1920x368/003791/ffffff",
          "name": "PlayStation Game A",
          "platform": "PS4",
        },
        Object {
          "cover": "https://via.placeholder.com/359x411/003791/ffffff",
          "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi in vehicula nulla. Nullam id leo non mi gravida iaculis. Integer aliquam ex in ipsum cursus, ac accumsan nisl cursus. Curabitur quis lorem sed enim ultrices scelerisque sit amet et sapien. Integer elementum, eros vel dignissim consectetur, nibh tortor iaculis leo, nec varius dolor velit in nisl. Quisque ac nibh quis risus venenatis mollis. Aliquam dapibus, mauris nec iaculis ornare, neque nulla lacinia turpis, ac luctus nunc tortor et est. Donec tempor lorem ac eros semper, eget ultricies urna ultrices. Nunc feugiat eros dui, id pharetra risus hendrerit sit amet.",
          "hero": "https://via.placeholder.com/1920x368/003791/ffffff",
          "name": "PlayStation Game B",
          "platform": "PS4",
        },
        Object {
          "cover": "https://via.placeholder.com/359x411/003791/ffffff",
          "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi in vehicula nulla. Nullam id leo non mi gravida iaculis. Integer aliquam ex in ipsum cursus, ac accumsan nisl cursus. Curabitur quis lorem sed enim ultrices scelerisque sit amet et sapien. Integer elementum, eros vel dignissim consectetur, nibh tortor iaculis leo, nec varius dolor velit in nisl. Quisque ac nibh quis risus venenatis mollis. Aliquam dapibus, mauris nec iaculis ornare, neque nulla lacinia turpis, ac luctus nunc tortor et est. Donec tempor lorem ac eros semper, eget ultricies urna ultrices. Nunc feugiat eros dui, id pharetra risus hendrerit sit amet.",
          "hero": "https://via.placeholder.com/1920x368/003791/ffffff",
          "name": "PlayStation Game C",
          "platform": "PS4",
        },
      ]
    }
  />,
]
`;
